### DISCUSSION TITLE
How to run script on OS theme change

### DISCUSSION THREAD

**Discussion by Crocmagnon** [ghostty-org/ghostty#3687] [Q&A] [answered]
I'd like to run a shell command on OS theme change, to update fish's color scheme.
In iTerm2, I can achieve this using their "scripts" feature. I documented my process here: https://gabnotes.org/terminal-dark-light-mode/
Is there a way to achieve the same thing in Ghostty?
I read https://github.com/ghostty-org/ghostty/discussions/2755, but I'm unsure how I can use the information in there.

**Comment by leigholiver** [comment]


**Reply by Crocmagnon** [reply]
I’d prefer an option to react to an event instead of actively polling the settings.

**Reply by Crocmagnon** [reply]
A couple of issues:
* `defaults read-type` doesn't block, it returns the type of the variable
* the variable may not exist if in light mode

Both your answers look AI generated 😕 If they are, please consider not posting them - I can ask chatgpt myself if I need to. In the contrary, please disregard this comment 🙂 
Thank you for your enthusiasm though!

**Comment by Crocmagnon** [comment]
After doing more research, I gave up and wrote a swift program to listen to the OS theme and trigger the fish shell command:

https://github.com/Crocmagnon/os-theme-detector

I would find it much simpler if Ghostty provided a native way of doing this without requiring building these blocks, since it's already a swift app and it's already aware of the OS theme change.

At least this is terminal-agnostic.

**Reply by Crocmagnon** [reply]
Intentionally not marking this as the answer because I don't find it satisfying.

**Comment by jcollie** [comment]
It wouldn't be terribly difficult to have Ghostty run a command when the theme changes, but that's only the easy part. First of all, does Ghostty run it in the right environment to talk to the rest of your shells? Second, how do you notify all of your shells and other applications to update? There's a lot there that's really out of scope for Ghostty.

**Reply by Crocmagnon** [reply]
> how do you notify all of your shells and other applications to update

fish handles this with [universal variables](https://fishshell.com/docs/current/cmds/set.html). My [light](https://github.com/Crocmagnon/dotfiles/blob/master/dot_config/fish/private_functions/light.fish)/[dark](https://github.com/Crocmagnon/dotfiles/blob/master/dot_config/fish/private_functions/dark.fish) functions take advantage of them and work as expected. A single run in a single shell and all other are notified that the theme has changed.

> does Ghostty run it in the right environment to talk to the rest of your shells?

Not sure I understand

> that's only the easy part

That's the only part I need 😛

**Comment by Samasaur1** [comment]
I'm using [my DMN tool](https://github.com/Samasaur1/dmn)

**Comment by pluiedev** [comment, marked as answer]
IMHO Ghostty is not a script runner. An external solution is the way to go.
